crates/sui/src/client_commands.rs

```
@@ -3098,3 +3107,66 @@ async fn check_protocol_version_and_warn(client: &SuiClient) -> Result<(), anyho
 
     Ok(())
 }
+
+/// Fetch move packages based on the provided package IDs.
+async fn fetch_move_packages(
+    client: &SuiClient,
+    package_ids: &[ObjectID],
+) -> Result<Vec<MovePackage>, anyhow::Error> {
+    let objects = client
```

What happens if one of the `package_ids` doesn't exist? Will we produce an error, or does it just get silently dropped? I think producing an error would be the safe thing to do in this case.

---

This will return an error when calling `into_object` below, if the object does not exist or is deleted. This is also covered by the changed output in one of the tests: `publish_with_nonexistent_dependency`.